base/src/Mp4WriterSink.cpp

```diff
@@ -249,6 +249,15 @@ bool DetailJpeg::write(frame_container& frames)
 		mp4_mux_sync(mux);
 		syncFlag = false;
 	}
+
+	if (!lastFrameTS)
```

Comment:
This same code block appears at 2 places... violates the DRY principal, 
consider using a private function

Reply:
Addressed it , created a function in the base class and using it.